Src/Coravel/Scheduling/Schedule/Event/ScheduledEvent.cs

```diff
@@ -46,6 +47,11 @@ public bool IsDue(DateTime utcNow)
 
         public async Task InvokeScheduledEvent()
         {
+            if(this.predicate != null && (!await predicate.Invoke())) 
```

Comment:
I think putting this in a private method ("WhenPrediatePasses") would read better?